src/unlimited.cpp

```diff
@@ -1705,8 +1705,11 @@ extern UniValue getstructuresizes(const UniValue &params, bool fHelp)
 
     ret.push_back(Pair("mapBlockIndex", mapBlockIndex.size()));
     // CChain
-    ret.push_back(Pair("setPreVerifiedTxHash", setPreVerifiedTxHash.size()));
-    ret.push_back(Pair("setUnVerifiedOrphanTxHash", setUnVerifiedOrphanTxHash.size()));
+    {
+        LOCK(cs_xval);
+        ret.push_back(Pair("setPreVerifiedTxHash", setPreVerifiedTxHash.size()));
```

Comment:
It's that guaranteed by the standard? I was thinking the implementation might have a pointer to an underlying object that returns the size and that that pointer might be reallocated by a modification in a separate thread.